lib/models/share.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:share/share.dart';
+
+class share {
+  //TODO: Put correct app link
+  String appLink = 'https://github.com/padamchopra/habito';
+  String _progress;
+
+  String getProgressString(List daysCompleted) {
+    return 'I have completed' + daysCompleted.toString();
+  }
+
+  void shareTheApp() {
+    Share.share('Check out Habito App at $appLink');
+  }
+
+  void sharePersonalProgress(List daysCompleted) {
+    _progress = getProgressString(daysCompleted);
+    Share.share(_progress);
+  }
```

Comment:
This should accept a String, and a List of String parameters. 
Make predefined string messages for different scenarios and use string interpolation to place parameters in them. 
Eg: Create public String messages in class like-
`String HABIT_COMPLETED_SUCCESSFULLY = "Hey, I've now developed a habit of ${paramsList.get(0)}"`
_important_ : Set the global `paramsList` = list obtained in parameter before using the string.